### ISSUE TITLE
Strict mode leftover? FFI in cairo_scaled_font_text_to_glyphs

### ISSUE THREAD

**Issue by Ducasse** [guillep/pharo-core#16914]
I was playing with Spec and Athens and the following example produces: 

<img width="411" alt="Capture 2024-07-17 at 18 04 41" src="https://github.com/user-attachments/assets/bc45b450-60fe-4136-be5f-cdd2de16bf17">

I found the following But I do not know how to fix it.

```
convertString: utf8String len: strlen ofFont: aScaledFont toGlyphs: glyphs numGlyphs: numGlyphs x: x y: y
	"cairo_status_t
	cairo_scaled_font_text_to_glyphs (cairo_scaled_font_t *scaled_font,
                                  double x,
                                  double y,
                                  const char *utf8,
                                  int utf8_len,
                                  cairo_glyph_t **glyphs,
                                  int *num_glyphs,
                                  cairo_text_cluster_t **clusters,
                                  int *num_clusters,
                                  cairo_text_cluster_flags_t *cluster_flags);"
	^ self ffiCall: #(
		cairo_status_t cairo_scaled_font_text_to_glyphs (CairoScaledFont aScaledFont,
			double x,
			double y,
			void * utf8String,
			int strlen,
			void ** glyphs,
			int * numGlyphs,
			void* 0,
			void* 0,
			void* 0))
```

Here is the code that I used (see below for the metacello snippet)
[AthensExamplePresenter.st.zip](https://github.com/user-attachments/files/16269299/AthensExamplePresenter.st.zip)

and 

```
AthensExamplePresenter new open
```
to reproduce it in Pharo12

**Comment by Ducasse** [comment]
When I hacked, the call worked so this is clearly a problem with the signature not following the strict mode declaration.

```
resolutionMode

	^ (self optionAt: #optStrict)
		ifTrue: [ FFIInferenceResolutionMode new ]
		ifFalse: [ FFIInferenceResolutionMode new ]
```

**Comment by Ducasse** [comment]
to load the code using iceberg

```
Metacello new
  baseline: 'CodeOfSpec20Book';
  repository: 'github://SquareBracketAssociates/CodeOfSpec20Book/src';
  load
```

**Comment by guillep** [comment]
We should probably backport https://github.com/pharo-project/pharo/issues/16562

**Comment by guillep** [comment]
This will be fixed with https://github.com/pharo-project/pharo/pull/16926/files

<blockquote><img src="https://avatars.githubusercontent.com/u/708322?s=400&v=4" width="48" align="right"><div><img src="https://github.githubassets.com/favicons/favicon.svg" height="14"> GitHub</div><div><strong><a href="https://github.com/pharo-project/pharo/pull/16926">Automatically type 'NULL'  arguments in FFI (backport to P12) by guillep · Pull Request #16926 · pharo-project/pharo</a></strong></div><div>Backport of #16562 to P12</div></blockquote>

**Comment by guillep** [comment]
Fixed with #16926
